innovation.game.php

```diff
@@ -9750,6 +9776,91 @@ function stInteractionStep() {
             );
             break;
         
+        // id 128, Artifacts age 2: Babylonian Chronicles
+        case "128C1A":
+            // "Transfer a top non-red card with a tower from your board to my board"
+            $options = array(
+                'player_id' => $player_id,
+                'n' => 1,
+                'can_pass' => false,
+                
+                'owner_from' => $player_id,
+                'location_from' => 'board',
+                'owner_to' => $launcher_id,
+                'location_to' => 'board',
+                
+                'with_icon' => 4, // tower
+                'color' => array(0,2,3,4) // non-red
+            );
+            break;
+
+        // id 131, Artifacts age 2: Holy Grail
+        case "131N1A":
+            // "Return a card from your hand"
+            $options = array(
+                'player_id' => $player_id,
+                'n' => 1,
+                'can_pass' => false,
+                
+                'owner_from' => $player_id,
+                'location_from' => 'hand',
+                'owner_to' => 0,
+                'location_to' => 'deck'
+            );
+            break;
+            
+        case "131N1B":
+            $age_selected = self::getGameStateValue('age_last_selected');
+            // "Claim an achievement of matching value, ignoring eligibility"
+            $options = array(
+                'player_id' => $player_id,
+                'n' => 1,
+                'can_pass' => false,
+                
+                'age' => $age_selected,
+                'owner_from' => 0,
+                'location_from' => 'achievements',
+                'owner_to' => $player_id,
+                'location_to' => 'achievements',
+
+                'require_achievement_eligibility' => false
+            );
+            break;
+
+
+        // id 132, Artifacts age 2: Terracotta Army
+        case "132C1A":
+            // "I compel you to return a top card with no tower"
+            $options = array(
+                'player_id' => $player_id,
+                'n' => 1,
+                'can_pass' => false,
+                
+                'owner_from' => $player_id,
+                'location_from' => 'board',
+                'owner_to' => 0,
+                'location_to' => 'deck',
+                
+                'without_icon' => 4 // tower
+            );
+            break;
+
+        case "132N1A":
+            // "Score a card from your hand with no towers"
+            $options = array(
+                'player_id' => $player_id,
+                'n' => 1,
+                'can_pass' => false,
+                
+                'owner_from' => $player_id,
+                'location_from' => 'hand',
+                'owner_to' => $player_id,
+                'location_to' => 'score',
+                
+                'without_icon' => 4 // tower
```

Comment:
Don't forget to add a line for 'score_keyword''! Or else this score won't count towards Monument.

Reply:
Thanks!

Reply:
done